game/layout/layout.py

```diff
@@ -0,0 +1,268 @@
+from __future__ import annotations
+
+import logging
+import random
+from dataclasses import dataclass, field
+from typing import TYPE_CHECKING, Type
+
+from dcs import Point
+from dcs.unit import Unit
+from dcs.unittype import UnitType as DcsUnitType
+
+from game.data.groups import GroupRole, GroupTask
+from game.data.units import UnitClass
+from game.point_with_heading import PointWithHeading
+from game.theater.theatergroundobject import (
+    SamGroundObject,
+    EwrGroundObject,
+    BuildingGroundObject,
+    MissileSiteGroundObject,
+    ShipGroundObject,
+    CarrierGroundObject,
+    LhaGroundObject,
+    CoastalSiteGroundObject,
+    VehicleGroupGroundObject,
+    IadsGroundObject,
+)
+from game.theater.theatergroup import TheaterUnit
+from game.utils import Heading
+
+if TYPE_CHECKING:
+    from game.factions.faction import Faction
+    from game.theater.theatergroundobject import TheaterGroundObject
+    from game.theater.controlpoint import ControlPoint
+
+
+class LayoutException(Exception):
+    pass
+
+
+@dataclass
+class LayoutUnit:
+    """The Position and Orientation of a single unit within the GroupLayout"""
+
+    name: str
+    position: Point
+    heading: int
+
+    @staticmethod
+    def from_unit(unit: Unit) -> LayoutUnit:
+        """Creates a LayoutUnit from a DCS Unit"""
+        return LayoutUnit(
+            unit.name,
+            Point(int(unit.position.x), int(unit.position.y)),
+            int(unit.heading),
+        )
+
+
+@dataclass
+class GroupLayout:
+    """The Layout of a TheaterGroup"""
+
+    name: str
+    units: list[LayoutUnit]
+
+    # The group this template will be merged into
+    group: int = 1
+
+    # Define the amount of random units to be created by the randomizer.
+    # This can be a fixed int or a random value from a range of two ints as tuple
+    unit_count: list[int] = field(default_factory=list)
+
+    # defintion which unit types are supported
+    unit_types: list[Type[DcsUnitType]] = field(default_factory=list)
+    unit_classes: list[UnitClass] = field(default_factory=list)
+    alternative_classes: list[UnitClass] = field(default_factory=list)
+
+    # Defines if this groupTemplate is required or not
+    optional: bool = False
+
+    # if enabled the specific group will be generated during generation
+    # Can only be set to False if Optional = True
+    enabled: bool = True
+
+    # TODO Caching for faction!
+    def possible_types_for_faction(self, faction: Faction) -> list[Type[DcsUnitType]]:
+        """TODO Description"""
+        unit_types = [t for t in self.unit_types if faction.has_access_to_dcs_type(t)]
+
+        alternative_types = []
+        for accessible_unit in faction.accessible_units:
+            if accessible_unit.unit_class in self.unit_classes:
+                unit_types.append(accessible_unit.dcs_unit_type)
```

Comment:
No use case so far to use both so thats why i did not add special handling for it. For me it's completly fine to have the results joined. Example "Class: Shorad, Unit_type: special AAA or MERAD Unit wich can also serve the specific task". This would allow to add specific units from other classes to the list without adding all units for this class.
As i said no use case so far and tbh i dont want to think about a use case which is not there yet. Whenever some one has a use case they should create an issue and then we can look into it closer. Not the other way around.